### DISCUSSION TITLE
Cypress.Cookies.defaults({ preserve: }) does not work in github actions

### DISCUSSION THREAD

**Discussion by inconduit** [cypress-io/cypress#24985] [Questions and Help]
i'm preserving an auth cookie across test files via cypress/support/e2e.js with Cypress.Cookies.defaults({ preserve:  })

my first test file does the authentication, visits an auth link it receives via email which sets a cookie on the client via the response.

i preserve that cookie with Cookies.defaults(), it gets sent in the request headers in subsequent tests.

it all works perfectly well when running cypress locally, either in the cypress ui, or in headless mode.

it does not work when running as a github action (cypress-io/github-action@v4). the cookie does not get preserved when the next test file is run.

is there anything i'm missing when running cypress through a github action?

**Comment by portyu9** [comment]
Hi @inconduit 

`Cypress.Cookies.defaults({ preserve: ... })` is no longer the supported solution. That API was removed, and the current Cypress mechanism for preserving authentication state is `cy.session()`.

For a login that should be reused across spec files, put the login in a shared command/helper and enable `cacheAcrossSpecs`:

```js
Cypress.Commands.add('login', () => {
  cy.session(
    'auth',
    () => {
      // Perform the login flow that creates your auth cookie.
    },
    {
      validate() {
        cy.getCookie('YOUR_AUTH_COOKIE').should('exist')
      },
      cacheAcrossSpecs: true,
    }
  )
})
```

Then call the login command from every spec that requires authentication:

```js
beforeEach(() => {
  cy.login()
  cy.visit('/')
})
```

One important CI limitation: `cacheAcrossSpecs: true` only shares the session **within the same `cypress run` on the same machine**. If your GitHub Actions workflow splits specs between separate jobs, containers, or parallel machines, browser/session state cannot be transferred between them. Each machine needs to establish its own authenticated session.

So I would first check whether the CI workflow is running those two specs in the same Cypress process/machine. The GitHub Action itself does not provide a separate cookie-preservation mechanism.

Please mark this answer as accepted if it helped, thank you.
